#! /usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t

gen = ParameterGenerator()

gen.add("kp_trans", double_t, 0, "kp_trans", 17., 0.0, 60)
gen.add("kd_trans", double_t, 0, "kd_trans", 11, 0.0, 60)
gen.add("ki_trans", double_t, 0, "ki_trans", 11., 0.0, 60)

gen.add("kp_angle", double_t, 0, "kp_angle", 8., 0.0, 30)
gen.add("kd_angle", double_t, 0, "kd_angle", 8, 0.0, 30)
gen.add("ki_angle", double_t, 0, "ki_angle", 5.7, 0.0, 30)

gen.add("history_length", int_t, 0, "history_length", 75, 0, 500)
gen.add("sub_mass", double_t, 0, "sub_mass", 9.7, 0.0, 100.0)
exit(gen.generate('subjugator_controller', 'pd_controller', 'Gain'))
